File: 71.简化路径.py
#
# @lc app=leetcode.cn id=71 lang=python3
#
# [71] 简化路径
#

# @lc code=start
class Solution:
    def simplifyPath(self, path: str) -> str:
        strs=path.split('/')
        result=[]
        for s in strs:
            if s=="" or s=='.':
                continue
            if s=='..' and len(result)!=0:
                result.pop()
            elif s!='..':
                result.append(s)
        res=''
        for s in result:
            res+='/'+s
        if len(result)==0:
            return '/'
        return res
# @lc code=end

File: 71.简化路径.cpp
/*
 * @lc app=leetcode.cn id=71 lang=cpp
 *
 * [71] 简化路径
 */

// @lc code=start
class Solution {
public:
    string simplifyPath(string path) {
        stringstream ss(path);
        string tmp="";
        vector<string> result;
        while(getline(ss,tmp,'/')){
            if(tmp==""||tmp==".")
                continue;
            if(tmp==".."&&!result.empty())
                result.pop_back();
            else if(tmp!="..")
                result.push_back(tmp);            
        }
        string res="";
        for(string str:result){
            res+="/"+str;
        }
        if(result.empty())
            return "/";
        return res;
    }
};
// @lc code=end

